src/run.ts

```diff
@@ -2,35 +2,48 @@ import { info, setFailed, getInput, warning } from "@actions/core";
 import { Context } from "@actions/github/lib/context";
 import { Jira } from "./jira";
 
-export const containsParentTaskID = async (ids: string[]): Promise<boolean> => {
+export const containsParentIssueID = async (ids: string[]): Promise<boolean> => {
   const jiraUrl = getInput("jiraUrl");
   if (!jiraUrl) {
-    warning(`Not checking parent task because of missing JIRA URL`);
+    warning(`Not checking parent issue because of missing JIRA URL`);
     return true;
   }
-  if (!ids) {
-    return false;
-  }
 
   const jira = new Jira(
     jiraUrl,
     getInput("jiraUsername"),
     getInput("jiraToken")
   );
 
-  const skipIDs = getInput("jiraSkipCheck").split(",");
-
   for (const id of ids) {
-    const issueID = id[0];
-    info(`Checking ID ${issueID}`);
-    if (skipIDs.includes(issueID.toLowerCase())) {
-      info(`Skipping ID ${issueID}`);
+    const issue = await jira.getIssue(id, { fields: "issuetype" });
+    if (issue.fields?.issuetype?.subtask === false) {
+      info(`Found parent issue ${id}`);
       return true;
     }
+  }
 
-    const issue = await jira.getIssue(issueID, { fields: "issuetype" });
-    if (issue.fields?.issuetype?.subtask === false) {
-      info(`Found parent task ${issueID}`);
+  return false;
+};
+
+export const isIssueSpike = async (ids: string[]): Promise<boolean> => {
+  const jiraUrl = getInput("jiraUrl");
+  if (!jiraUrl) {
+    warning(`Not checking parent issue because of missing JIRA URL`);
+    return false;
+  }
+
+  const jira = new Jira(
+    jiraUrl,
+    getInput("jiraUsername"),
+    getInput("jiraToken")
+  );
+
+  for (const id of ids) {
+    const issue = await jira.getIssue(id, { fields: "issuetype" });
+    if (issue.fields?.issuetype?.subtask === false &&
```

Comment:
why is there a check for subtask in a function isIssueSpike?

Reply:
to make sure the issue is not a subtask, im not sure what are the use cases where the issue can be both a Spike and a subtask so i left the subtask check there as well just in case

Reply:
an issue is either a subtask or a spike, it can't be both.